arches/app/utils/index_database.py

```diff
@@ -262,7 +262,7 @@ def index_resources_using_singleprocessing(
                     bar = pyprind.ProgBar(resource_count, bar_char="█", title=title)
                 else:
                     bar = None
-
+            batch_size = batch_size if batch_size > 8 else 8
             for resource in optimize_resource_iteration(
                 resources, chunk_size=batch_size // 8
```

Comment:
Let's make this a new variable so we don't confuse it with the elastic batch size.
```suggestion
            chunk_size = max(batch_size // 8, 8)
            for resource in optimize_resource_iteration(
                resources, chunk_size=chunk_size
```